tools/options.py

```diff
@@ -82,6 +82,8 @@ def get_default_options_parser(add_clean=True, add_options=True,
         parser.add_argument("--app-config", default=None, dest="app_config",
                             type=argparse_filestring_type,
                             help="Path of an app configuration file (Default is to look for 'mbed_app.json')")
+    parser.add_argument("--coverage-filter", default=[], action="append", dest="coverage_filter",
```

Comment:
I'd like to see a useful message displayed along the lines of 
"Code coverage not supported with compiler xxxxx"